src/loader/mod.rs

```diff
@@ -258,9 +261,15 @@ mod tests {
     fn test_cmdline_write_end() {
         let gm = create_guest_mem();
         let mut cmdline_address = GuestAddress(45);
+
         let mut cl = Cmdline::new(10);
-        cl.insert_str("1234").unwrap();
+        cl.insert_str("123").unwrap();
```

Comment:
Why this change?

Reply:
The actual change that I wanted to introduce was to catch the overwriting of the cmdline in guest's memory such that the byte of the null terminator (before overwriting to memory) not to be 0. This way I could make sure that the cmdline really contains the null terminator that is written to memory.

The change of the string from "1234" to "123" was just to reduce the number of bytes to check from the memory (and also check the entire cmdline that was written into memory). Definitely, I could have checked just the bytes around the null terminator or maybe use totally different bytes for the overwriting but it was not something more complex than that.

Reply:
I change the first cmdline that is written into memory to contain "123" and the second one to contain "45". Doing like this, when calling load_cmdline() with the same starting address in memory, there is no risk to make confusions between which byte was written by which call of the load_cmdline().